### DISCUSSION TITLE
Socket.io should probably ESM export "Server"

### DISCUSSION THREAD

**Discussion by richardeschloss** [socketio/socket.io#4499] [Ideas] [answered]
https://github.com/socketio/socket.io/blob/945c84be47d2923a9132786c9fd11dd90fa2c6db/lib/index.ts#L1002

I think the ESM export should include Server too? It seems like it's only exported with the CommonJS. Some modern frameworks may not work well with the mixed CJS / ESM exports, and may throw an error when trying to `import { Server } from "socket.io"`

Related issue: https://github.com/richardeschloss/nuxt-socket-io/issues/278

I appreciate your help!

**Comment by darrachequesne** [comment, marked as answer]
Hi! The `Server` class is actually exported: https://github.com/socketio/socket.io/blob/945c84be47d2923a9132786c9fd11dd90fa2c6db/lib/index.ts#L102

But there might be an issue in the underlying `engine.io` package: https://github.com/socketio/engine.io/issues/657

**Reply by richardeschloss** [reply]
Ok, will close this in favor of https://github.com/socketio/engine.io/issues/657
